lang/LangPrimSource/PyrStringPrim.cpp

```diff
@@ -568,12 +568,18 @@ int prString_Getenv(struct VMGlobals* g, int /* numArgsPushed */) {
         return err;
 
 #ifdef _WIN32
-    char buf[1024];
-    DWORD size = GetEnvironmentVariable(key, buf, 1024);
-    if (size == 0 || size > 1024)
-        value = 0;
-    else
-        value = buf;
+    auto key_w = win32_CharToWchar(key);
+    DWORD size = GetEnvironmentVariableW(key_w.get(), nullptr, 0);
+    auto value_w = std::make_unique<wchar_t[]>(size);
+    size = GetEnvironmentVariableW(key_w.get(), value_w.get(), size);
+
+    std::unique_ptr<char[]> value_ptr;
```

Comment:
I've defined `value_ptr` here so that it stays in scope until `PyrString* pyrString = newPyrString(g->gc, value, 0, true);`. Without this, if I did `value = win32_WcharToChar(value_w.get()).get()` (line 580), I'd get garbage data in `value`....

Reply:
Hi, Since you commented on this section, I’m assuming you're not certain about this code?